### PR TITLE
fix(inventory): default value if value not exist in inventory

### PR DESCRIPTION
**Pull request by stonebuzz**
Prevent PHP warning
```shell
[2022-11-04 12:10:33] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "frequency" in /home/stanislas/TECLIB/DEV/GLPI/10.0-bugfixes/src/Inventory/Asset/Device.php at line 145
  Backtrace :
  src/Inventory/Asset/MainAsset.php:862              Glpi\Inventory\Asset\Device->handle()
  src/Inventory/Asset/MainAsset.php:782              Glpi\Inventory\Asset\MainAsset->handleAssets()
  src/RuleImportAsset.php:950                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1510                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1591                        Rule->process()
  src/Inventory/Asset/MainAsset.php:564              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:702                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:337                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:90                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:305    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:242    Glpi\Agent\Communication\AbstractRequest->handleXMLRequest()
  front/inventory.php:92                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
```

When inventory file not provide (not mandatory) information
Example ```CPUS``` without ```speed``` (glpi ```frequency```)

```xml
    <CPUS>
      <ARCH>i386</ARCH>
      <CORE>1</CORE>
      <FAMILYNAME>Other</FAMILYNAME>
      <FAMILYNUMBER>6</FAMILYNUMBER>
      <ID>63 06 00 00 FD FB 8B 07</ID>
      <MANUFACTURER>Intel</MANUFACTURER>
      <MODEL>6</MODEL>
      <NAME>QEMU Virtual CPU version 2.4.0</NAME>
      <STEPPING>3</STEPPING>
      <THREAD>1</THREAD>
    </CPUS>
```

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: src/Inventory/Asset/Processor.php [hunk 1]
```diff
@@ -59,7 +59,10 @@ public function prepare(): array
             }
             if (property_exists($val, 'frequency')) {
                 $val->frequency_default = $val->frequency;
-                $val->frequence = $val->frequency;
+                $val->frequency = $val->frequency;
```
Comment (cedric-anne) [inline src/Inventory/Asset/Processor.php:62]: In database, property is named `frequence`. Are you sure this property has to be renamed ?

Comment (trasher) [reply src/Inventory/Asset/Processor.php:62]: Field is named `frequence` in `glpi_deviceprocessors` table, but `frequency` in `glpi_items_deviceprocessors`...

Comment (trasher) [reply src/Inventory/Asset/Processor.php:62]: Anyways, that part of the fix seems indeed wring and should be removed. Missing data is fixed in the else part

File: src/Inventory/Asset/Processor.php [hunk 2]
```diff
@@ -59,7 +59,10 @@ public function prepare(): array
...
+            } else {
+                $val->frequency_default = 0;
+                $val->frequency = 0;
```
Comment (trasher) [inline src/Inventory/Asset/Processor.php:65]: Maybe also:
```suggestion
                $val->frequency = 0;
                $val->frequence = 0;
```

File: src/Inventory/Asset/Processor.php (hunk 1 above)
Comment (trasher) [reply src/Inventory/Asset/Processor.php:62]:
```suggestion
                $val->frequence = $val->frequency;
```

File: src/Inventory/Asset/Processor.php [hunk 3]
```diff
@@ -60,6 +60,10 @@ public function prepare(): array
             if (property_exists($val, 'frequency')) {
                 $val->frequency_default = $val->frequency;
                 $val->frequence = $val->frequency;
+            } else {
+                $val->frequency_default = 0;
+                $val->frequency = 0;
```
Comment (cedric-anne) [inline src/Inventory/Asset/Processor.php:65]:
```suggestion
```

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**